kong/tools/utils.lua

```diff
@@ -1468,4 +1468,66 @@ do
 end
 _M.time_ns = time_ns
 
+
+local sha256_bin
+do
+  local digest = require "resty.openssl.digest"
+  local sha256_digest
+
+  function sha256_bin(key)
+    local _, bin, err
+    if not sha256_digest then
+      sha256_digest, err = digest.new("sha256")
+      if err then
+        return nil, err
+      end
+    end
+
+    bin, err = sha256_digest:final(key)
+    if err then
```

Comment:
should we add a `reset` here?

Reply:
yes I guess it would be a good idea to start fresh in case of errors